File: src/easy/plus-minus.ts
export const plusMinus = (target: number[]): number[] => {
    const length = target.length;
    const [left, right] = target.reduce(
        (accum: number[], num: number) => {
            if (num > 0) accum[0] += 1;
            if (num < 0) accum[1] += 1;
            return accum;
        },
        [0, 0],
    );
    const l = left / length;
    const r = right / length;
    const middle = 1 - (l + r);
    return [l.toFixed(3), r.toFixed(3), middle.toFixed(3)].map(Number);
};

console.log(plusMinus([0, 0, -1, 1, 1]), [0.4, 0.2, 0.4]);

File: easy/plus_minus.py
def plus_minus(sample):
    length = len(sample)
    score = [0, 0]
    for num in sample:
        if num > 0:
            score[0] += 1
        if num < 0:
            score[1] += 1
    left = score[0] / length
    middle = score[1] / length
    right = 1 - (left + middle)
    return [left, middle, right]
